### DISCUSSION TITLE
Medusa + Stripe and the secret

### DISCUSSION THREAD

**Discussion by GSdaBlessedFist** [medusajs/medusa#5885] [Q&A]
I'm working with a NExt.js 14 frontend trying to integrate Stripe.  In the documentation I find for the `medusa-payment-stripe`, they provide:  
```
//medusa-config.js  (serverside)
{
    resolve: `medusa-payment-stripe-kj`,
    options: {
      api_key: process.env.STRIPE_API_KEY,
      webhook_secret: process.env.STRIPE_WEBHOOK_SECRET,
    },
  }
```
Stripe provides a Publishable key and a Secret key.
```
//.env (serverside)
STRIPE_API_KEY=pk_test_51K5....
STRIPE_WEBHOOK_SECRET=sk_test_51....
```
When I attempt checking out and get to the Payment section, the system stalls.
```
serverside console:
Error:....
 type: 'invalid_data',
  code: 'secret_key_required',  <----
```
Where do we find this secret key and is it the same as the WEBHOOK_SECRET?

**Comment by srindom** [comment]
The server env variable named `STRIPE_API_KEY` should be the secret key you obtain from the Stripe dashboard (the one that starts with `sk`). The publishable key (the one that starts with `pb`) should be configured in your storefront.

The `WEBHOOK_SECRET` is obtained from the Stripe dashboard after you have created a webhook in the dashboard.
